test/integration/main_test.go

```diff
@@ -0,0 +1,416 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/iam/apiv1/iampb"
+	"github.com/abcxyz/access-on-demand/pkg/cli"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/sethvargo/go-retry"
+	"google.golang.org/genproto/googleapis/type/expr"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	resourcemanager "cloud.google.com/go/resourcemanager/apiv3"
+)
+
+var (
+	cfg           *config
+	projectClient *resourcemanager.ProjectsClient
+)
+
+func TestMain(m *testing.M) {
+	os.Exit(func() int {
+		ctx := context.Background()
+
+		if strings.ToLower(os.Getenv("TEST_INTEGRATION")) != "true" {
+			log.Printf("skipping (not integration)")
+			// Not integration test. Exit.
+			return 0
+		}
+
+		// set up global test config.
+		c, err := newTestConfig(ctx)
+		if err != nil {
+			log.Printf("Failed to parse integration test config: %v", err)
+			return 2
+		}
+		cfg = c
+
+		// Set up global resource manager client.
+		pc, err := resourcemanager.NewProjectsClient(ctx)
+		if err != nil {
+			log.Printf("failed to create projects client: %v", err)
+			return 2
+		}
+		defer pc.Close()
+		projectClient = pc
+
+		return m.Run()
+	}())
+}
+
+func TestIAMHandle(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name         string
+		filePath     string
+		wantBindings []*iampb.Binding
+		wantOutput   string
+	}{
+		{
+			name:     "success",
+			filePath: fmt.Sprintf("../../%s/iam.yaml", cfg.WorkingDir),
+			wantBindings: []*iampb.Binding{
+				{
+					Role:    "roles/actions.Viewer",
+					Members: []string{"user:aod-test-bot@tycho.joonix.net"},
+					Condition: &expr.Expr{
+						Title:      cfg.ConditionTitle,
+						Expression: fmt.Sprintf("request.time < timestamp('%s')", cfg.IAMExpiration),
+					},
+				},
+				{
+					Role:    "roles/ml.viewer",
+					Members: []string{"user:aod-test-bot@tycho.joonix.net"},
+					Condition: &expr.Expr{
+						Title:      cfg.ConditionTitle,
+						Expression: fmt.Sprintf("request.time < timestamp('%s')", cfg.IAMExpiration),
+					},
+				},
+			},
+			wantOutput: fmt.Sprintf(`------Successfully Handled IAM Request------
+iamrequest:
+  policies:
+    - resource: projects/access-on-demand-i-12af76
+      bindings:
+        - members:
+            - user:aod-test-bot@tycho.joonix.net
+          role: roles/actions.Viewer
+        - members:
+            - user:aod-test-bot@tycho.joonix.net
+          role: roles/ml.viewer
+duration: %sh0m0s
+starttime: %s
+`, cfg.IAMExpirationDurationHour, cfg.IAMExpirationStartTime),
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			args := []string{
+				"iam", "handle",
+				"-path", tc.filePath,
```

Comment:
How would this work? The files have `${INTEG_TEST_PROJECT_ID}` in them.

Instead of having those template files, maybe inline them and write test file to https://pkg.go.dev/testing#T.TempDir

Reply:
[envsubst](https://github.com/abcxyz/access-on-demand/compare/main...sqin/ci#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR107) works. 

In order to use envsubst, we need to have those template files.

Reply:
Why bother to use envsubst if you can already do that in code?

Reply:
We will have to do that for cleanup file anyway, as it is handled in workflow.

Reply:
I'm not sure I understand the reasoning.